libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -391,14 +387,20 @@ void AC_AutoTune::control_attitude()
         test_run(axis, direction_sign);
 
         // Check for failure causing reverse response
-        if (lean_angle <= -AUTOTUNE_TARGET_MIN_ANGLE_RLLPIT_CD) {
+        if (lean_angle <= -attitude_control->lean_angle_max_cd() * AUTOTUNE_TARGET_MIN_ANGLE_RLLPIT) {
```

Comment:
I wonder if we should have this:

```suggestion
        if (lean_angle <= -lean_angle_max_cd()) {
```

Reply:
No, this is where the aircraft is going in the opposite direction to what it should be going. We don't want to wait for it to get all the way out to lean angle max. This threshold can and should be small.

Reply:
Sorry, I wasn't clear in my comment.

I was suggesting breaking out a method:
```
float lean_angle_max() const {
    return attitude_control->lean_angle_max_cd() * AUTOTUNE_TARGET_MIN_ANGLE_RLLPIT;
}
```

Most of the lines touched in this patch add that multiplication, so make a method for it.

Reply:
I would offer another suggestion. Could we make this a virtual method so that heli and multi can apply different criteria to reverse response and exceeding max lean angle.  I left these alone when i first implemented heli autotune but now I don't know if I can make these changes work for heli, especially the 2/3 max lean angle criteria.